modules/study/src/main/BSONHandlers.scala

```diff
@@ -305,20 +307,42 @@ object BSONHandlers {
     implicit def CrazyDataBSONHandler: BSON[PocketData] =
       new BSON[PocketData] {
         private def writePocket(p: Pocket) = p.roles.map(_.forsyth).mkString
-        //will need to be updated for different GameFamilys
-        private def readPocket(p: String) = Pocket(
-          p.view.flatMap(r => Role.forsyth(variant.gameLogic, r)).toList
-        )
         def reads(r: Reader) =
-          PocketData.Chess(
-            strategygames.chess.PocketData(
-              promoted = r.getsD[strategygames.chess.Pos]("o").toSet,
-              pockets = Pockets(
-                p1 = readPocket(r.strD("w")),
-                p2 = readPocket(r.strD("b"))
+          variant.gameLogic match {
+            case GameLogic.Chess() =>
+              PocketData.Chess(
+                chess.PocketData(
+                  promoted = Set(),
```

Comment:
`promoted` needs to be set to what it was for chess. I dont think its relevant for any of the other gamelogics but it also wouldn't hurt to add in equivalent lines for them.